tests/test_directives.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) 2017 The Regents of the University of Michigan
+# All rights reserved.
+# This software is licensed under the BSD 3-Clause License.
+import sys
+
+from flow import Directives, _DirectivesItem, OnlyType
+
+
+def raise_if_too_small(value):
+    def is_greater(v):
+        try:
+            if v < value:
+                raise ValueError
+        except (TypeError, ValueError):
+            raise ValueError("Expected a number greater than {}. Received {}"
+                             "".format(value, v))
+        return v
+
+
+def _evaluate(value, job=None):
+    if callable(value):
+        if job is None:
+            raise RuntimeError("job not specified when evaluating directive.")
+        else:
+            return value(job)
+    else:
+        return value
+
+
+def finalize_np(value, directives):
+    nranks = directives.get('nranks', 1)
+    omp_num_threads = directives.get('omp_num_threads', 1)
+    if callable(nranks) or callable(omp_num_threads):
+        return value
+    else:
+        return max(value, max(1, nranks) * max(1, omp_num_threads))
+
+
+def is_fraction(value):
+    if 0 <= value <= 1:
+        return value
+    else:
+        raise ValueError("Value must be beween 0 and 1.")
+
+
+def no_aggregation(v, o):
+    return v
+
+
+nonnegative_int = OnlyType(int, postprocess=raise_if_too_small(-1))
+natural_number = OnlyType(int, postprocess=raise_if_too_small(0))
+nonnegative_real = OnlyType(float, postprocess=raise_if_too_small(0))
+
+
+class TestDirectivesBase():
```

Comment:
There must have been a miscommunication. There is no need to have a `FlowProject` for these tests. What we want to test is the `Directives` and `_DirecttivesItem` classes themselves without a project.

Reply:
Okay then I will create instances directly without a `mock_project`.